src/utils/spo.ts

```diff
@@ -1574,5 +1693,105 @@ export const spo = {
     };
 
     await request.post(requestOptions);
+  },
+
+  /**
+   * Retrieves the ObjectIdentity from a SharePoint site
+   * @param webUrl web url
+   * @param logger The logger object
+   * @param verbose If in verbose mode
+   * @return The ObjectIdentity as string
+   */
+  async requestObjectIdentity(webUrl: string, logger: Logger, verbose: boolean): Promise<string> {
+    const res = await spo.getRequestDigest(webUrl);
+    const formDigestValue = res.FormDigestValue;
+
+    const requestOptions: CliRequestOptions = {
+      url: `${webUrl}/_vti_bin/client.svc/ProcessQuery`,
+      headers: {
+        'X-RequestDigest': formDigestValue
+      },
+      data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><Query Id="1" ObjectPathId="5"><Query SelectAllProperties="false"><Properties><Property Name="ServerRelativeUrl" ScalarProperty="true" /></Properties></Query></Query></Actions><ObjectPaths><Property Id="5" ParentId="3" Name="Web" /><StaticProperty Id="3" TypeId="{3747adcd-a3c3-41b9-bfab-4a64dd2f1e0a}" Name="Current" /></ObjectPaths></Request>`
+    };
+
+    const response = await request.post<string>(requestOptions);
+    if (verbose) {
+      logger.logToStderr('Attempt to get _ObjectIdentity_ key values');
+    }
+
+    const json: ClientSvcResponse = JSON.parse(response);
+
+    const contents: ClientSvcResponseContents = json.find(x => { return x['ErrorInfo']; });
+    if (contents && contents.ErrorInfo) {
+      throw contents.ErrorInfo.ErrorMessage || 'ClientSvc unknown error';
+    }
+
+    const identityObject = json.find(x => { return x['_ObjectIdentity_']; });
```

Comment:
Couldn't we refer to `_ObjectIdentity_` as a property rather than as indexer, so that we can get design-time type safety?